File: offgame-ui/src/app/model/TechnologyDetails.ts
import {Requirement} from "./Requirement";

export class TechnologyDetails {

    description!: string;
    imageName!: string;
    name!: string;
    requirements: Requirement[];

    constructor(data: any) {
        Object.assign(this, data);

        this.requirements = data.requirements.map((object: any) => new Requirement(object))
    }
}

File: offgame-core/src/main/java/org/retal/offgame/dto/TechnologyDetails.java
package org.retal.offgame.dto;

import lombok.*;

import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TechnologyDetails {

    private String name;
    private String imageName;
    private String description;
    private List<RequirementDTO> requirements;
}
